src/main/java/com/blamejared/contenttweaker/color/ColorHandler.java

```diff
@@ -0,0 +1,36 @@
+package com.blamejared.contenttweaker.color;
+
+import com.blamejared.contenttweaker.VanillaFactory;
+import com.blamejared.contenttweaker.api.blocks.IIsCoTBlock;
+import com.blamejared.contenttweaker.api.functions.IBlockColorSupplier;
+import com.blamejared.contenttweaker.api.functions.IItemColorSupplier;
+import com.blamejared.contenttweaker.api.items.IIsCotItem;
+import com.blamejared.crafttweaker.impl.item.MCItemStack;
+import net.minecraft.block.Block;
+import net.minecraft.util.IItemProvider;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.ColorHandlerEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
+public class ColorHandler {
+    @SubscribeEvent
+    public static void onItemColor(ColorHandlerEvent.Item event) {
+        event.getItemColors().register((stack, tintIndex) ->
+                        VanillaFactory.REGISTRY.getFunction(((IIsCotItem) stack.getItem()), IItemColorSupplier.class)
+                                .map(iItemColorSupplier -> iItemColorSupplier.apply(new MCItemStack(stack), tintIndex))
+                                .orElse(-1),
+                VanillaFactory.REGISTRY.getItemsAsVanillaItems().toArray(IItemProvider[]::new));
+    }
+
+    @SubscribeEvent
+    public static void onBlockColor(ColorHandlerEvent.Block event) {
+        event.getBlockColors().register((state, world, pos, tintIndex) ->
```

Comment:
You should not register for blocks that do not have color handlers

Reply:
You are right. But I make color handlers can reload. But register events are execured before CrT script are loaded. If you think it has some problem. I can add a `colorized` method to BlockBuilder, which make only colorized blocks will register their color hanlers. But it is same as `tickRandomly` things above.

Reply:
The issue with this is in the past other mods have registered color handlers for things, like foods. And it breaks when you have an already registered handlers even if it returns -1